core/src/main/java/hudson/triggers/Trigger.java

```diff
@@ -156,7 +158,7 @@ public TriggerDescriptor getDescriptor() {
      * periodically. This is useful when your trigger does
      * some polling work.
      */
-    protected Trigger(String cronTabSpec) throws ANTLRException {
+    protected Trigger(@Nonnull String cronTabSpec) throws ANTLRException {
```

Comment:
They just shouldn't need to be set. - the tooling should cope with this when it does null escape analysis - and adding this in IMO just makes the code even harder to scan.
IntelliJ should be clever enough here, eclipse is clever enough that I have to turn it off (Jenkins produces thousands of warnings!) - so rather should focus on what could be null and add annotations there (much less should be null tollerant than not null tollerant)